diff -rupN boost_1_45_0.orig/tools/build/v2/tools/darwin.jam boost_1_45_0/tools/build/v2/tools/darwin.jam
--- boost_1_45_0.orig/tools/build/v2/tools/darwin.jam	2010-07-12 16:37:43.000000000 +0900
+++ boost_1_45_0/tools/build/v2/tools/darwin.jam	2010-12-06 16:32:53.000000000 +0900
@@ -423,7 +423,7 @@ rule setup-address-model ( targets * : s
         
         case arm :
         {
-            options = -arch armv6 ;
+            options = -arch armv7 -mthumb -mcpu=cortex-a8 -mfloat-abi=softfp -mfpu=neon ;
         }        
     }
     
